### ISSUE TITLE
Bug: Snowflake: Autocomplete: Only shows distinct table names

### ISSUE THREAD

**Issue by jim256** [hankinsoft/SQLPro#464]
If I have DATABASE_1.SCHEMA_1 and DATABASE_1.SCHEMA_2, who both have a table named "PRODUCTS" and I type:
`SELECT
	*
	
FROM
	PRODU`

I only get the suggestion for the PRODUCTS table from one of the schemas, even though there is a table with that name in each schema.

macOS 10.15.5 (19F101)
SQLPro Studio 2020.61 (Build 110689.5)
Snowflake 4.22.1

**Comment by jim256** [comment]
Just this since I'm still seeing it and it makes it so I sometimes can't use autocomplete.

SQLPro Studio 2021.13 (Build 110775.5)
macOS 11.2.1 (20D74)
Snowflake 5.5.0

**Comment by hankinsoft** [comment]
Fixed for the next build (2021.19). Submitted and should be available within ~24 hours.

**Comment by jim256** [comment]
Awesome thanks!
Seems to be working well.

One more thing this made me realize is that, if I have a table whose exact name is contained in another table name, I won't get a suggestion for the longer if I type the shorter.
Example:
I have one table named "product" and another named "product_attributes". If I type "product", I don't get the suggestion for "product_attributes". Not sure which would be the better UX, but I'd personally lean toward still offering the suggestions for the other objects, especially since I sometimes might just time "product" when I want the actual "product" table, but want autocomplete to fill in the potentially-long schema name for me.

Happy to make a separate issue for this if you feel that makes more sense.

**Comment by hankinsoft** [comment]
Hmmm, I'm not able to reproduce this with either columns or table names.

<img width="1386" alt="Screen Shot 2021-03-04 at 1 45 23 PM" src="https://user-images.githubusercontent.com/3160528/110006360-ef5fe800-7cef-11eb-8440-f8adf4b34e19.png">
<img width="1386" alt="Screen Shot 2021-03-04 at 1 41 56 PM" src="https://user-images.githubusercontent.com/3160528/110006362-ef5fe800-7cef-11eb-8a51-4342a61cb727.png">

**Comment by jim256** [comment]
What about with Snowflake?

**Comment by hankinsoft** [comment]
So I'm still not able to reproduce this.

<img width="1382" alt="Screen Shot 2021-03-18 at 10 43 11 AM" src="https://user-images.githubusercontent.com/3160528/111635962-cc94ff80-87d6-11eb-88c1-fc6681682231.png">

Are you still seeing this in the latest build? And am I understanding this correctly? (Is my above example doing what you are saying)?

**Comment by jim256** [comment]
I apologize - I'm not sure if I was on an old build or something, but this seems to be working great in 2021.27 (Build 110784.5)
Thanks a lot!

**Comment by hankinsoft** [comment]
Are you still seeing this? I've tried to reproduce again with no luck. If you are still seeing this, would it be possible to post a video? I'm pretty sure I understand what your explaining but again I'm not able to see it, so there is a change I'm misunderstanding.

**Comment by jim256** [comment]
I think the original issue is gone. But the last issue, for which I reopened this, is still present.

Before the following screenshots, I created two tables by executing:

```
CREATE OR REPLACE TABLE PUBLIC.HANKINSOFT AS
	
	SELECT
		1 INT
;


CREATE OR REPLACE TABLE PUBLIC.HANKINSOFT_TABLE AS
	
	SELECT
		1 INT
;
```

<img width="359" alt="image" src="https://user-images.githubusercontent.com/15026225/119047062-a7f60900-b97a-11eb-8e5d-c82a9de7f416.png">
If I type `FROM HANKINSOF`, I get two suggestions.

<img width="210" alt="image" src="https://user-images.githubusercontent.com/15026225/119047095-b6dcbb80-b97a-11eb-9944-5b83d723420a.png">
If I type out `FROM HANKINSOFT`, I get no suggestions, because I've typed the exact name of a table (this is what I'd like to see changed).

<img width="359" alt="image" src="https://user-images.githubusercontent.com/15026225/119047156-c9ef8b80-b97a-11eb-83a0-980bdab7aafd.png">

If I type `FROM HANKINSOFT_`, I get one suggestion.

Attached is a very clean log file.
[com.hankinsoft.osx.sqlprostudio 2021-05-20--20-54-23-340.log](https://github.com/hankinsoft/SQLPro/files/6518719/com.hankinsoft.osx.sqlprostudio.2021-05-20--20-54-23-340.log)


SQLPro Studio 2021.53 (Build 110808.5)
macOS 11.3.1 (20E241)
Snowflake 5.19.1

**Comment by hankinsoft** [comment]
From the log file, I think I see what's going on - your inserting the completion of a partial word to insert HANKINSOFT? If you have HANKINSOFT entered and hit escape to manually trigger autocomplete,  you see both entries?

What is your autocomplete delay set to?

I think the underlying issue is that autocomplete does not appear anytime a 'completion has been forced completed' (by hitting enter or tab). But if you could answer the above two questions that would confirm it for me.

**Comment by jim256** [comment]
I think the narrative I provided is the most helpful.

I'm not using Esc or manually triggering autocomplete in any way.

The issue is simply that if I have two tables, one named `THIS_TABLE`, and another that has the same name but with anything appended (such as `THIS_TABLE_DATA`, `THIS_TABLE_IS BETTER`, `THIS_TABLE_HAS_DATA`, etc.), and I type `FROM THIS_TABLE`, I will not receive any autocomplete suggestions. I believe this is because autocomplete recognizes that I've already typed the exact name of a table, so it doesn't think I need any suggestions.
